### PR TITLE
Oleksijt task15 user permision control without class

### PR DESCRIPTION
**Pull request by Oleksijt**
Does this solution meet the requirements of the task?
Would it be better then previous solution?

### CODE REVIEW

File: oleksijt/users.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            if args[0].role == 'admin':
+                return func(*args, **kwargs)
+            else:
+                raise PermissionError
+        except PermissionError:
+            print('Access denied. You can not perform this operation.')
+    return wrapper
+
+
+class User:
```
[the remark this answers is not in the document]
Comment (Oleksijt) [reply oleksijt/users.py]: Here - https://github.com/pivancy/programming_education/pull/68/commits/b71cbfda6dd6b3b78ff6728ef0a8e5142bc95e0f

File: oleksijt/users.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
+""" This module contents functions and classes, that serve Users:
```
Comment (pivancy) [inline oleksijt/users.py]: Why do we need this file here?

Comment (Oleksijt) [reply oleksijt/users.py]: I forgot to remove it.
Fixed.

File: oleksijt/users_without_class.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+`check_perms` - checks permission for user. This function
+     decorated by access_control function.
+"""
+
+admin_name = 'Oleksiy'
+
+
+def access_control(func):
+    # Decorator function which controls user access to functions
+    def wrapper(name):
```
Comment (pivancy) [inline oleksijt/users_without_class.py]: Why do not write something like that?

    def access_control(fn):
        def wrapped(*args, **kwargs):
            if get_user() != admin_name:
                raise PermissionException("Permission denied.")
            return fn(*args, **kwargs)
        return wrapped

Comment (Oleksijt) [reply oleksijt/users_without_class.py]: Done.
When should i use Try: Except:?

Comment (pivancy) [reply oleksijt/users_without_class.py]: You don't need try except here.

Comment (Oleksijt) [reply oleksijt/users_without_class.py]: Ok. I did it without "except".
But can you explain in what cases i need it?

Comment (pivancy) [reply oleksijt/users_without_class.py]: This functionality is super useful for the platforms with limited access for different groups of users.

Comment (pivancy) [reply oleksijt/users_without_class.py]: It would be great to make this decorator possible to use in this way:

Comment (Oleksijt) [reply oleksijt/users_without_class.py]: Done.

File: oleksijt/users_without_class.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    # Function checks permission for user.
+    return 'Allowed access'
+
+
+if __name__ == '__main__':
+    print(get_user_info('oleksijt'))  # Prints `Oleksiy`
+    print(get_user_info('Yuriy'))  # Prints `Permission denied.`
+
+    print(check_perms('pivanchy'))  # Prints `Allowed access`
+    print(check_perms('Yuriy'))  # Prints `Permission denied.`
```
Comment (pivancy) [inline oleksijt/users_without_class.py]: Why this returns Permission denied if Yuriy is in moderators list?

Comment (Oleksijt) [reply oleksijt/users_without_class.py]: Fixed.

File: oleksijt/users_without_class.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    return access_control
+
+
+@allowed_users(admins_list)
+def get_user_info(name):
+    # Returns  user's name.
+    return name
+
+
+@allowed_users(admins_list + moderators_list)
```
Comment (pivancy) [inline oleksijt/users_without_class.py]: It's not a good idea for adding two lists in this way.

File: oleksijt/users_without_class.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+admins_list = ['pivanchy', 'oleksijt']
+moderators_list = ['Yuriy', 'oleksijt']
+
+
+# Decorator function which controls user access to functions
+def allowed_users(user_list):
+    def access_control(func):
+        def wrapper(*args, **kwargs):
+            if args[0] not in user_list:
+                raise PermissionError('Permission denied.')
```
Comment (pivancy) [inline oleksijt/users_without_class.py]: I suggest you to rename it to `PermissionDenied` and define this custom Exception to avoid a `NameError`.

**Review by UkrainianEngineer — CHANGES_REQUESTED** [review]

File: oleksijt/users_without_class.py (hunk 5 above)
Comment (Oleksijt) [reply oleksijt/users_without_class.py]: Fixed. Now it ads only unique values.

File: oleksijt/users_without_class.py (hunk 6 above)
Comment (Oleksijt) [reply oleksijt/users_without_class.py]: Done.

File: oleksijt/users_without_class.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    return 'Allowed access'
+
+
+if __name__ == '__main__':
+    print(get_user_info('oleksijt'))  # Prints `Oleksiy`
+    print(get_user_info('pivanchy'))  # Prints `pivanchy`
+    try:
+        print(get_user_info('Yuriy'))  # Prints `Permission denied.`
+    except PermissionDenied:
+        print('Permission denied.')
```
Comment (pivancy) [inline oleksijt/users_without_class.py]: I don't think you really need to handle it here, but ok.

**Review by UkrainianEngineer — APPROVED** [review]

**Comment by pivancy** [comment]
👍

File: oleksijt/users_without_class.py (hunk 7 above)
Comment (Oleksijt) [reply oleksijt/users_without_class.py]: I've handled it to see further testing result without error.

**Pull request merged**